apps/cyberstorm-remix/app/p/tabs/Versions/PackageVersionVersions.tsx

```diff
@@ -0,0 +1,175 @@
+import "./Versions.css";
+import {
+  NewTableSort,
+  NewTable,
+  NewTableLabels,
+  Heading,
+  SkeletonBox,
+  NewLink,
+} from "@thunderstore/cyberstorm";
+import { Await, LoaderFunctionArgs } from "react-router";
+import { useLoaderData } from "react-router";
+import { DapperTs } from "@thunderstore/dapper-ts";
+import semverGt from "semver/functions/gt";
+import semverLt from "semver/functions/lt";
+import {
+  TableCompareColumnMeta,
+  TableRow,
+} from "@thunderstore/cyberstorm/src/newComponents/Table/Table";
+import {
+  getPublicEnvVariables,
+  getSessionTools,
+} from "cyberstorm/security/publicEnvVariables";
+import { Suspense } from "react";
+import { isSemver } from "cyberstorm/utils/typeChecks";
+import { DownloadLink, InstallLink, ModManagerBanner } from "./common";
+
+export async function loader({ params }: LoaderFunctionArgs) {
+  if (params.communityId && params.namespaceId && params.packageId) {
+    const publicEnvVariables = getPublicEnvVariables(["VITE_API_URL"]);
+    const dapper = new DapperTs(() => {
+      return {
+        apiHost: publicEnvVariables.VITE_API_URL,
+        sessionId: undefined,
+      };
+    });
+    return {
+      communityId: params.communityId,
+      namespaceId: params.namespaceId,
+      packageId: params.packageId,
+      versions: dapper.getPackageVersions(params.namespaceId, params.packageId),
+    };
+  }
+  return {
+    status: "error",
+    message: "Failed to load versions",
+    versions: [],
+  };
+}
+
+export async function clientLoader({ params }: LoaderFunctionArgs) {
+  if (params.communityId && params.namespaceId && params.packageId) {
+    const tools = getSessionTools();
+    const dapper = new DapperTs(() => {
+      return {
+        apiHost: tools?.getConfig().apiHost,
+        sessionId: tools?.getConfig().sessionId,
+      };
+    });
+    return {
+      communityId: params.communityId,
+      namespaceId: params.namespaceId,
+      packageId: params.packageId,
+      versions: dapper.getPackageVersions(params.namespaceId, params.packageId),
+    };
+  }
+  return {
+    status: "error",
+    message: "Failed to load versions",
+    versions: [],
+  };
+}
+
+function rowSemverCompare(
+  a: TableRow,
+  b: TableRow,
+  columnMeta: TableCompareColumnMeta
+) {
+  if (isSemver(String(a[0].sortValue)) && isSemver(String(b[0].sortValue))) {
+    if (semverLt(String(a[0].sortValue), String(b[0].sortValue))) {
+      return columnMeta.direction;
+    }
+    if (semverGt(String(a[0].sortValue), String(b[0].sortValue))) {
+      return -columnMeta.direction;
+    }
+  }
+  return 0;
+}
```

Comment:
Would this work? or do we not have semverCompare?
Also I think this function should probably be a more generic helper the comparison itself isn't really PackageVersion specific.

```
function rowSemverCompare(
  a: TableRow,
  b: TableRow,
  columnMeta: TableCompareColumnMeta
) {
  const av = String(a[0].sortValue);
  const bv = String(b[0].sortValue);

  if (isSemver(av) && isSemver(bv)) {
    return semverCompare(av, bv) * columnMeta.direction;
  }

  return 0;
}
```

Or even just call `semverCompare(av, bv) * columnMeta.direction;` direclty inline without the wrapper function if that works?